packages/app-headless-cms/src/admin/config/contentEntries/list/Browser/Filter.tsx

```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import { Property, useIdGenerator } from "@webiny/react-properties";
+import { useModel } from "~/admin/hooks";
+
+export interface FilterConfig {
+    name: string;
+    element: React.ReactElement;
+}
+
+export interface FilterProps {
+    name: string;
+    element?: React.ReactElement<unknown>;
+    modelIds?: string[];
+    remove?: boolean;
+    before?: string;
+    after?: string;
+}
+
+export const Filter: React.FC<FilterProps> = ({
+    name,
+    element,
+    modelIds = [],
+    after = undefined,
+    before = undefined,
+    remove = false
+}) => {
+    const { model } = useModel();
+    const getId = useIdGenerator("filter");
+
+    if (modelIds.length > 0 && !modelIds.includes(model.modelId)) {
+        return null;
+    }
+
+    const placeAfter = after !== undefined ? getId(after) : undefined;
```

Comment:
I think you can make the `useIdGenerator()` to allow the `undefined` value as input of the function it returns, and then just return `undefined` from that function in case when the input is `undefined`.
This way you will need to have a lot of checks before using the `getId()`.